src/D2L.Bmx/Okta/OktaSessionStorage.cs

```diff
@@ -0,0 +1,54 @@
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using D2L.Bmx.Okta.Models;
+
+namespace D2L.Bmx.Okta;
+
+internal class OktaSessionStorage {
+
+	internal static void SaveSessions( List<OktaSessionCache> sessions ) {
+
+		string jsonString = JsonSerializer.Serialize(
+			sessions.ToArray(),
+			SourceGenerationContext.Default.OktaSessionCacheArray );
+		File.WriteAllText( SessionsFileName(), jsonString );
+	}
+
+	internal static List<OktaSessionCache> Sessions() {
+
+		var bmxDirectory = BmxDir();
+		var sessionsFileName = SessionsFileName();
+
+		if( !Directory.Exists( bmxDirectory ) ) {
+			Directory.CreateDirectory( bmxDirectory );
+		}
+
+		if( !File.Exists( sessionsFileName ) ) {
+			File.WriteAllText( sessionsFileName, "[]" );
+		}
+
+		try {
+			var sessionsJson = File.ReadAllText( SessionsFileName() );
+			var sessions = JsonSerializer.Deserialize( sessionsJson, SourceGenerationContext.Default.OktaSessionCacheArray );
+			if( sessions is not null ) {
+				return sessions.ToList();
+			}
+		} catch( JsonException ) {
+			return new();
+		}
+		return new();
+	}
+
+	internal static string SessionsFileName() {
```

Comment:
For consistency's sake could you refactor all uses of .bmx directory ex: https://github.com/Brightspace/bmx/blob/4d2e39358594fca24b30f9af28f3a068c408e9f1/src/D2L.Bmx/BmxConfigProvider.cs#L14

To either 1) All pull from one place that defines the folder paths or 2) do it inline in the BmxConfigProvider

Reply:
Seems like the only other use is the one you mentioned